test/my_solver.jl

```diff
@@ -0,0 +1,62 @@
+struct MySolver <: AbstractBlockSolver
+    MySolver() = begin
+        @warn "This solver does not provide dual variable information."
+        new()
+    end
+end
+
+mutable struct PGDSolverInstance
```

Comment:
Does it have to be a `mutable struct`? I would prefer using just `struct` if possible

Reply:
In this case, we need it to be a `mutable struct.` I am using this to initialize some arrays that are updated in every iteration of the gradient descent algorithm.

Reply:
Ok then!